CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc

```diff
@@ -65,10 +65,10 @@ AlignPCLThresholdsWriter::AlignPCLThresholdsWriter(const edm::ParameterSet& iCon
       m_minNrecords(iConfig.getParameter<unsigned int>("minNRecords")),
       m_parameters(iConfig.getParameter<std::vector<edm::ParameterSet> >("thresholds")) {
   //now do what ever initialization is needed
-  myThresholds = new AlignPCLThresholds();
+  AlignPCLThresholds myThresholds{};
```

Comment:
```suggestion
```
Out of my "jurisdiction", but I couldn't understand this line. Afaiu, this is creating a local variable `myThresholds` in the ctor scope, which is ["shadowing"](https://stackoverflow.com/questions/30494407/declaring-a-local-variable-within-class-scope-with-same-name-as-a-class-attribut) the class member `myThresholds`; the latter isn't getting modified, and the local variable gets destroyed in the very next line. Related to this, it might be clearest to rename `myThresholds` (class member) to `m_thresholds`. No?

Reply:
As it is now this line is useless. Before when it was a pointer it was `new`ed in the constructor.

Reply:
This was the link Marco showed me earlier on why the `myThresholds{}` is better than `myThresholds`
https://stackoverflow.com/questions/18222926/why-is-list-initialization-using-curly-braces-better-than-the-alternatives

Reply:
Yes, @tvami  but that does not apply here,  this is a class member not a local variable put in the stack in the analyze method (which is where I used that syntax)

Reply:
Ah ok!
Btw for class members I think CMS prefers `myThresholds_` instead of `m_thresholds`, no?
2.9 in https://cms-sw.github.io/cms_coding_rules.html

Reply:
So for my understanding what is the suggestion here?
- remove this line alltogether and use the class memeber directly (leaving an "empty" ctor)?
- make it a local variable in the stak in the analyze method and remove both this line and the class member?

Reply:
Sure, it's just that that plugin happens to call its members `m_*`, and it's maybe clearer to be consistent (one could of course rename all of them).

Reply:
> make it a local variable in the stak in the analyze method and remove both this line and the class member?

I didn't even realise this was an option (if so, sounds like the better one to me). In any case, I leave suggestions to others (I don't know this plugin).

Reply:
> I didn't even realise this was an option (if so, sounds like the better one to me). 

IIUC from the code `myThresholds` is only modified and used in the analyze method, so in principle we could directly do everything there.

>In any case, I leave suggestions to others (I don't know this plugin).

Same for me, but @mmusich is the original author and main maintiner of this plugin so I leave the final decision to him.

Reply:
Either way is fine to me, whatever takes less typing (likely just remove the line).

Reply:
I was curious to test all possibilities so I did several tests already :)
I will push in a minute the final one with: I moved `myThresholds` from being a class member to be an instance created in the analyze method.

Reply:
done in 41e9574